File: basic-math/average/average.go
package main

import "fmt"

func average(numbers []int) float64 {
	var sum int
	for _, number := range numbers {
		sum += number
	}
	return float64(sum) / float64(len(numbers))
}

func main() {
	numbers := []int{1, 2, 3, 4, 5}
	avg := average(numbers)
	fmt.Printf("The average of %v is %f\n", numbers, avg)
}

File: basic-maths/average/average.ts
function average(numbers: number[]): number {
  if (numbers.length === 0) {
    throw new Error("Array cannot be empty");
  }
  const sum = numbers.reduce((acc, num) => acc + num, 0);
  return sum / numbers.length;
}

console.log(average([1, 2, 3, 4, 5]));
console.log(average([10, 20, 30, 40, 50]));